torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -202,6 +198,10 @@ void ProcessGroupAgent::join() {
       SendWork(allWorkerInfo_[dst], Message({}, {}, MessageType::SHUTDOWN)));
   threadPool_.waitWorkComplete();
   listenerThread_.join();
```

Comment:
is it intentional to join listenerThread_ and cleanup PythonRpcHandler in both join and shutdown?

Reply:
Right now, the functionality is in both functions though it will execute only once, since the intended use case is to only call `rpc.shutdown()` or `rpc.wait_all_workers()` for a clean shutdown that waits for other workers.